cou/steps/plan.py

```diff
@@ -491,6 +525,60 @@ def _get_archive_data_steps(analysis_result: Analysis, args: CLIargs) -> list[Pr
     return []
 
 
+def _get_set_noout_steps(analysis_result: Analysis, args: CLIargs) -> list[PreUpgradeStep]:
+    """Get set noout steps.
+
+    :param analysis_result: Analysis result
+    :type analysis_result: Analysis
+    :param args: CLI arguments
+    :type args: CLIargs
+    :return: List of pre-upgrade steps.
+    :rtype: list[PreUpgradeStep]
+    """
+    if args.set_noout and args.upgrade_group in {DATA_PLANE, None}:
+        return [
+            PreUpgradeStep(
+                description="Set ceph cluster 'noout' flag before data plane upgrade",
+                coro=ceph.ensure_noout(analysis_result.model, True),
```

Comment:
This ensure_noout is a little bit confusing to me.

From here I would assume that you want to set noout and in the function but the docstring says:
```
:param unset: True to unset noout, False to set noout
```

Wouldn't need to change here to actually set `noout` to False?

I think it would be less confusing if you use `set`instead of `unset`

Reply:
updated, hope it's clearer